content/rosa/entra-external-auth/index.md

```diff
@@ -0,0 +1,467 @@
+oc apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: rosa-admins
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: $GROUP_ID
+EOF
+```
+
+### Validate External Authentication
+
+#### Validate using the OpenShift Web Console
+
+To validate you can log into the OpenShift console with Microsoft Entra ID, first retrieve the OpenShift Console URL by running the following command:
+
+```bash
+rosa describe cluster -c ${ROSA_CLUSTER_NAME} -o json | jq -r '.console.url'
+```
+
+Open this URL and your external authentication provider should open.
+
+### Validate access to the cluster's API
+Here we will be leveraging a client-go Credential Plugin to automate the interaction between the OpenShift Command Line Interface and the External Authentication Provider including the management of OIDC tokens.
+
+More information related to Kubernetes plugins can be found [here](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/).
+
+Once the plugin has been configured and is available, create a KUBECONFIG file here with EntraID details for example create **rosa-auth.kubeconfig** file with following information
+
+```bash
+apiVersion: v1
+clusters:
+- cluster:
+    server: ${API_URL}
+  name: cluster
+contexts:
+- context:
+    cluster: cluster
+    namespace: default
+    user: oidc
+  name: admin
+current-context: admin
+kind: Config
+preferences: {}
+users:
+- name: oidc
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - oidc-login
+      - get-token
+      - --oidc-issuer-url=https://login.microsoftonline.com/${TENANT_ID}/v2.0
+      - --oidc-client-id=${CLIENT_ID}
+      - --oidc-client-secret=${CLIENT_SECRET}
+      - --oidc-extra-scope=email
+      - --oidc-extra-scope=openid
+      command: kubectl
+      env: null
+      interactiveMode: Never
+```
+Set the KUBECONFIG environment variable to the location of the rosa-cluster.kubeconfig file which will enable the OpenShift Command Line Interface to authenticate against the ROSA cluster with the contents of the file.. 
+
+To confirm that the identity details from the OIDC token is being used by the OpenShift Command Line Interface:
+
+```bash
+export KUBECONFIG=$(pwd)/rosa-auth.kubeconfig
+```
+
+Confirm access to cluster with elevated priviledges
+
+```bash
+oc get nodes
+```
+example output:
+
+```bash
+NAME                         STATUS   ROLES    AGE     VERSION
+ip-10-0-0-170.ec2.internal   Ready    worker   3h29m   v1.30.7
+ip-10-0-1-171.ec2.internal   Ready    worker   3h30m   v1.30.7
+ip-10-0-2-161.ec2.internal   Ready    worker   3h29m   v1.30.7
+```
+To verify logged in as user of group run
+
+```bash
+oc auth whoami
+```
+### Validate access to the cluster's API using device-code authentication 
+
+Note: This option is only used when the client machine lacks keyring. 
```

Comment:
This is inaccurate. Using device-code authentication is when you are authenticating from somewhere that doesn't have a web browser. The `--token-cache-storage=disk` parameter is used when a client machine lacks a keyring.